### DISCUSSION TITLE
Merge multiple Textures

### DISCUSSION THREAD

**Discussion by tguichaoua** [bevyengine/bevy#3222] [Q&A] [answered]
I followed the example of [shader/array_texture](https://github.com/bevyengine/bevy/blob/3de391be21666c90661066ca395c63813352b4bb/examples/shader/array_texture.rs) that require a Texture that contains the differents textures.

Is there a way to load the texture from several files, to create the texture at runtime ?

**Comment by MrGVSV** [comment, marked as answer]
You may want to check out the [`TextureAtlasBuilder`](https://github.com/bevyengine/bevy/blob/3de391be21666c90661066ca395c63813352b4bb/examples/2d/texture_atlas.rs).

Specifically, the merging happens here:
https://github.com/bevyengine/bevy/blob/3de391be21666c90661066ca395c63813352b4bb/examples/2d/texture_atlas.rs#L51-L57
And it can be used to create a texture of textures, so to speak

**Reply by tguichaoua** [reply]
I just tried it, but `TextureAtlasBuilder` arange texture in a square and texture array required the textures to be aranged vertically.
In the `texture_atlas` example, I tried to load tiles from `assets/rpg/tiles` (each tile is 16x16) and set the `max_size` of the `TextureAtlasBuilder` to `Vec2::new(16.0, f32::INFINITY)`. But I got the `NotEnoughSpace` error.
It's possible to force texture to be arranged vertically ?

**Reply by MrGVSV** [reply]
In that case (where every texture is a tile of the exact same size), you can check out a crate I made specifically for tilesets called [`bevy_tile_atlas`](https://github.com/MrGVSV/bevy_tile_atlas). It has a similar API as `TextureAtlasBuilder` but it auto-resizes indefinitely and you can set a maximum number of columns.

```rust
use bevy::prelude::*;
use bevy_tile_atlas::TileAtlasBuilder;

/// Creates a tile-based, ordered `TextureAtlas`
///
/// Assumes that the given handles are all loaded and in their desired order
fn build_tileset(handles: Vec<Handle<Texture>>, textures: &mut Assets<Texture>) -> TextureAtlas {
  let mut builder = TileAtlasBuilder::default().max_columns(1usize);
  
  for handle in handles {
    let texture = textures.get(&handle).unwrap();
    builder.add_texture(handle, texture);
  }
  
  builder.finish(textures).unwrap()
}
```
